From 4fc7634fb2b94142425f80ae42e3b9046d397a68 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Thu, 7 Aug 2025 23:19:22 +0100
Subject: [PATCH 1/5] Adapt build scripts for LeOS

---
 generate.sh | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/generate.sh b/generate.sh
index 8f24152..283664f 100644
--- a/generate.sh
+++ b/generate.sh
@@ -15,8 +15,8 @@ fi
 echo 'PRODUCT_MAKEFILES := \' > AndroidProducts.mk
 
 for part in a ab;do
-	for apps in vanilla gapps foss gapps-go old;do
-		for arch in arm64 arm a64;do
+	for apps in vanilla;do
+		for arch in arm64 a64;do
 			for su in yes no;do
 				apps_suffix=""
 				apps_script=""
@@ -86,6 +86,7 @@ for part in a ab;do
 TARGET_GAPPS_ARCH := ${baseArch}
 include build/make/target/product/aosp_${baseArch}.mk
 \$(call inherit-product, device/phh/treble/base.mk)
+\$(call inherit-product, vendor/LeOS/leos.mk)
 $optional_base
 $apps_script
 $rom_script
@@ -94,7 +95,7 @@ PRODUCT_NAME := $target
 PRODUCT_DEVICE := tdgsi_${arch}_$part
 PRODUCT_BRAND := google
 PRODUCT_SYSTEM_BRAND := google
-PRODUCT_MODEL := TrebleDroid $apps_name
+PRODUCT_MODEL := LeOS-A16
 
 # Overwrite the inherited "emulator" characteristics
 PRODUCT_CHARACTERISTICS := device
-- 
2.49.0
