From cd678c8c78577539a3e91aa9089432916bae4db4 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Tue, 5 Aug 2025 09:48:47 +0100
Subject: [PATCH 2/4] Enable adb debug and disable secure

---
 rw-system.sh | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/rw-system.sh b/rw-system.sh
index 3e25c0b..0bddcb9 100644
--- a/rw-system.sh
+++ b/rw-system.sh
@@ -833,8 +833,8 @@ if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ];then
     resetprop_phh ro.boot.veritymode enforcing
     resetprop_phh ro.boot.warranty_bit 0
     resetprop_phh ro.warranty_bit 0
-    resetprop_phh ro.debuggable 0
-    resetprop_phh ro.secure 1
+    resetprop_phh ro.debuggable 1
+    resetprop_phh ro.secure 0
     resetprop_phh ro.build.type user
     resetprop_phh ro.product.build.type user
     resetprop_phh ro.system.build.type user
@@ -842,8 +842,6 @@ if [ -f /system/phh/secure ] || [ -f /metadata/phh/secure ];then
     resetprop_phh ro.vendor.build.type user
     resetprop_phh --delete ro.build.selinux
 
-    resetprop_phh ro.adb.secure 1
-
     # Hide system/xbin/su
     mount /mnt/phh/empty_dir /system/xbin
     mount /mnt/phh/empty_dir /system/app/me.phh.superuser
-- 
2.49.0
