From 3a8f61222cb61bceccc2f15cd7522178971a1e2e Mon Sep 17 00:00:00 2001
From: harvey186 <harvey186@hotmail.com>
Date: Sun, 10 Nov 2024 09:25:19 +0100
Subject: [PATCH] BUILD_BROKEN_MISSING_REQUIRED_MODULES

---
 leos_a64_ab/BoardConfig.mk   |  2 ++
 leos_arm64_ab/BoardConfig.mk |  2 ++


diff --git a/leos_a64_ab/BoardConfig.mk b/leos_a64_ab/BoardConfig.mk
index f131e88..551bb8f 100644
--- a/leos_a64_ab/BoardConfig.mk
+++ b/leos_a64_ab/BoardConfig.mk
@@ -8,3 +8,5 @@ ifeq ($(BOARD_SYSTEMIMAGE_PARTITION_RESERVED_SIZE),)
 BOARD_SYSTEMIMAGE_PARTITION_SIZE := 1572864000
 endif
 TARGET_USES_64_BIT_BINDER := true
+
+BUILD_BROKEN_MISSING_REQUIRED_MODULES := true

diff --git a/leos_arm64_ab/BoardConfig.mk b/leos_arm64_ab/BoardConfig.mk
index 2fddc52..25f632a 100644
--- a/leos_arm64_ab/BoardConfig.mk
+++ b/leos_arm64_ab/BoardConfig.mk
@@ -6,3 +6,5 @@ BOARD_SYSTEMIMAGE_PARTITION_SIZE := 2147483648
 else
 BOARD_SYSTEMIMAGE_PARTITION_RESERVED_SIZE := 33554432
 endif
+
+BUILD_BROKEN_MISSING_REQUIRED_MODULES := true
-- 
2.34.1
