From 4fc7634fb2b94142425f80ae42e3b9046d397a68 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Thu, 7 Aug 2025 23:19:22 +0100
Subject: [PATCH 1/1] Adapt build scripts for LeanOS

---
 generate.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/generate.sh b/generate.sh
index 8fc94a7..29fcd54 100644
--- a/generate.sh
+++ b/generate.sh
@@ -95,7 +95,7 @@ PRODUCT_NAME := $target
 PRODUCT_DEVICE := tdgsi_${arch}_$part
 PRODUCT_BRAND := google
 PRODUCT_SYSTEM_BRAND := google
-PRODUCT_MODEL := TrebleDroid $apps_name
+PRODUCT_MODEL := LeanOS

 # Overwrite the inherited "emulator" characteristics
 PRODUCT_CHARACTERISTICS := device
--
2.49.0
